Arm_Software/Arm_Software.ino

```diff
@@ -0,0 +1,356 @@
+    {
+       driveTo  = Joint.pid.incrementPid(moveTo, Angle, 2.5);
+    }
+    output = driveTo;
+}
+
+void moveToAngle(RoveJointDifferential &Joint, float tiltTo, float twistTo, float Angles[2], float outputs[2])
+{
+    float tilt = 0;
+    float twist = 0;
+    float smallerAngle = 0;
+    float largerAngle =  0;
+    float fakeTilt = 0;
+    float fakeTiltAngle = 0;
+    float fakeTwist = 0;
+    float fakeTwistAngle = 0;
+    ///MATH FOR J1
+    //check if it's faster to go from 360->0 or 0->360 then the normal way
+    smallerAngle = min(twistTo, Angles[1]);
+    largerAngle =  max(twistTo, Angles[1]);
+    //if wrapping around 360 is faster than going normally
+    if((smallerAngle+(360000-largerAngle)) < abs(Angles[1]-twistTo))
+    {
+      if(Angles[1]-(smallerAngle+(360000-largerAngle))<0)
+      {
+        fakeTwist  = Angles[1]+(smallerAngle+(360000-largerAngle));
+        fakeTwistAngle = fakeTwist-((360000-twistTo) + (Angles[1]-0));
+       
+      }
+      else if(Angles[1]-(smallerAngle+(360000-largerAngle))>0)
+      {
+        fakeTwist  = Angles[1]-(smallerAngle+(360000-largerAngle));
+        fakeTwistAngle = fakeTwist+((twistTo-0) + (360000-Angles[1]));
+       
+      }
+      twist  = -Joint.twistPid.incrementPid(fakeTwist, fakeTwistAngle,2.5);
+  
+    }
+    else if((smallerAngle+(360000-largerAngle)) >= abs(Angles[1]-twistTo))
+    {
+       twist  = Joint.twistPid.incrementPid(twistTo, Angles[1],2.5);
+    }
+
+    ///MATH FOR J0
+    //check if it's faster to go from 360->0 or 0->360 then the normal way
+    smallerAngle = min(tiltTo, Angles[0]);
+    largerAngle =  max(tiltTo, Angles[0]);
+    //if wrapping around 360 is faster than going normally
+    if((smallerAngle+(360000-largerAngle)) < abs(Angles[0]-tiltTo))
+    {
+       if(Angles[0]-(smallerAngle+(360000-largerAngle))<0)
+      {
+        fakeTilt  = Angles[0]+(smallerAngle+(360000-largerAngle));
+        fakeTiltAngle = fakeTilt-((360000-tiltTo) + (Angles[0]-0));
+       
+      }
+      else if(Angles[0]-(smallerAngle+(360000-largerAngle))>0)
+      {
+        fakeTilt  = Angles[0]-(smallerAngle+(360000-largerAngle));
+        fakeTiltAngle = fakeTilt+((tiltTo-0) + (360000-Angles[0]));
+       
+      }
+      tilt  = -Joint.tiltPid.incrementPid(fakeTilt, fakeTiltAngle,2.5);
+    }
+    //if the normal way is faster, or equal we want less of a headache
+    else if((smallerAngle+(360000-largerAngle)) >= abs(Angles[0]-tiltTo))
+    {
+       tilt  = Joint.tiltPid.incrementPid(tiltTo, Angles[0],2.5);
+    }
+
+    outputs[0] = tilt;
+    outputs[1] = twist;
+}
+
+void telemetry()
+{
+    RoveComm.write(RC_ARMBOARD_JOINTANGLES_DATA_ID, RC_ARMBOARD_JOINTANGLES_DATA_COUNT, jointAngles);
+}
+
+
+void estop()
+{
+    ShoulderTilt.DriveMotor(0);
+    ShoulderTwist.DriveMotor(0);
+    ElbowTilt.DriveMotor(0);
+    ElbowTwist.DriveMotor(0);
+    Wrist.tiltTwistDrive(0, 0);
+    digitalWrite(LaserToggle, LOW);
```

Comment:
Might be worth discussing if the laser should be turned off during estop()? This would kill the laser the second the arm isn't being moved for ~100 milliseconds right?